aptos-move/block-executor/src/view.rs

```diff
@@ -1889,4 +1895,135 @@ mod test {
+        assert_eq!(unsync_map.fetch_data(&KeyType::<u32>(3, false)), Some((Arc::new(TransactionWrite::from_state_value(Some(state_value_3))), None)));
+        
+        let patched_value = Value::struct_(Struct::pack(vec![Value::struct_(Struct::pack(vec![
+            Value::u64(5),
+            Value::u64(30),
+        ]))]));
+        let state_value_4 = StateValue::new_legacy(patched_value.simple_serialize(&layout).unwrap().into());
+        assert_eq!(latest_view.get_resource_state_value(&KeyType::<u32>(4, false), Some(&layout)).unwrap(), Some(state_value_4.clone()));
+        assert!(latest_view.read_set_sequential_execution().borrow().contains(&KeyType(4, false)));
+        assert_eq!(unsync_map.fetch_data(&KeyType::<u32>(4, false)), Some((Arc::new(TransactionWrite::from_state_value(Some(state_value_4))), Some(Arc::new(layout)))));
+    }
+
+    #[test]
+    fn test_read_operations_parallel() {
+        let counter = AtomicU32::new(5);
+        let state_value_3 = StateValue::new_legacy(Bytes::from(Value::u64(12321).simple_serialize(&MoveTypeLayout::U64).unwrap()));
+        let mut data = HashMap::new();
+        data.insert(
+            KeyType::<u32>(3, false),
+            state_value_3.clone(),
+        );
+        let layout = MoveTypeLayout::Struct(MoveStructLayout::new(vec![MoveTypeLayout::Struct(
+            MoveStructLayout::new(vec![
+                MoveTypeLayout::Tagged(
+                    LayoutTag::IdentifierMapping(IdentifierMappingKind::Aggregator),
+                    Box::new(MoveTypeLayout::U64),
+                ),
+                MoveTypeLayout::U64,
+            ]),
+        )]));
+        let value = Value::struct_(Struct::pack(vec![Value::struct_(Struct::pack(vec![
+            Value::u64(25),
+            Value::u64(30),
+        ]))]));
+        let state_value_4 = StateValue::new_legacy(value.simple_serialize(&layout).unwrap().into());
+        data.insert(
+            KeyType::<u32>(4, false),
+            state_value_4,
+        );
+        let base_view = MockStateView::new(data);
+        let versioned_map = MVHashMap::new();
+        let scheduler = Scheduler::new(10);
+        let latest_view = LatestView::<TestTransactionType, MockStateView, MockExecutable>::new(
+            &base_view,
+            super::ViewState::Sync(super::ParallelState::new(
+                &versioned_map,
+                &scheduler,
+                &counter
+            )),
+            1,
+        );
+
+        assert_eq!(latest_view.get_resource_state_value(&KeyType::<u32>(1, false), None).unwrap(), None);
+        assert_eq!(latest_view.get_resource_state_value(&KeyType::<u32>(2, false), Some(&layout)).unwrap(), None);
+        assert_eq!(latest_view.get_resource_state_value(&KeyType::<u32>(3, false), None).unwrap(), Some(state_value_3.clone()));
+       
+        //TODO: This is printing Ok(Versioned(Err(StorageVersion), ValueType { bytes: Some(b"!0\0\0\0\0\0\0"), metadata: None }, None))
```

Comment:
Having `ValueType { bytes: Some(b"!0\0\0\0\0\0\0"), metadata: None }, None)` seems wrong, here we should have bytes from `state_value_3`?

Reply:
I think the value is null because the version is Err(StorageVersion)? Not sure if this is the expected behaviour or a bug.